src/main/java/org/jabref/gui/openoffice/OpenOfficePanel.java

```diff
@@ -359,93 +335,91 @@ private void exportEntries() {
+        dialogPane.setContent(indicator);
+        FXDialog progressDialog = dialogService.showCustomDialog(Localization.lang("Autodetecting paths..."), dialogPane, ButtonType.CANCEL);
+        progressDialog.show();
 
         try {
+            ooPrefs = jabRefPreferences.getOpenOfficePreferences();
             // Add OO JARs to the classpath
-            loadOpenOfficeJars(Paths.get(preferences.getInstallationPath()));
+            loadOpenOfficeJars(Paths.get(ooPrefs.getInstallationPath()));
 
-            // Show progress dialog:
-            progressDialog = new DetectOpenOfficeInstallation(diag, preferences, dialogService)
-                    .showProgressDialog(diag, Localization.lang("Connecting"), Localization.lang("Please wait..."));
-            JDialog finalProgressDialog = progressDialog;
             BackgroundTask
-                    .wrap(this::createBibBase)
-                    .onFinished(() -> SwingUtilities.invokeLater(() -> {
-                        finalProgressDialog.dispose();
-                        diag.dispose();
-                    }))
-                    .onSuccess(ooBase -> {
-                        this.ooBase = ooBase;
-
-                        if (ooBase.isConnectedToDocument()) {
-                            frame.output(Localization.lang("Connected to document") + ": " + ooBase.getCurrentDocumentTitle().orElse(""));
-                        }
-
-                        // Enable actions that depend on Connect:
-                        selectDocument.setEnabled(true);
-                        pushEntries.setEnabled(true);
-                        pushEntriesInt.setEnabled(true);
-                        pushEntriesEmpty.setEnabled(true);
-                        pushEntriesAdvanced.setEnabled(true);
-                        update.setEnabled(true);
-                        merge.setEnabled(true);
-                        manageCitations.setEnabled(true);
-                        exportCitations.setEnabled(true);
-
-                    })
-                    .onFailure(ex ->
-                            dialogService.showErrorDialogAndWait(Localization.lang("Autodetection failed"), Localization.lang("Autodetection failed"), ex))
-                    .executeWith(Globals.TASK_EXECUTOR);
-            diag.dispose();
+                          .wrap(this::createBibBase)
+                          .onFinished(() -> progressDialog.close())
+                          .onSuccess(ooBase -> {
+                              this.ooBase = ooBase;
+
+                              if (ooBase.isConnectedToDocument()) {
+                                  dialogService.notify(Localization.lang("Connected to document") + ": " + ooBase.getCurrentDocumentTitle().orElse(""));
+                              }
+
+                              // Enable actions that depend on Connect:
+                              selectDocument.setDisable(false);
```

Comment:
introduce a boolean property and bind the visibility property to it? Reduces code complexity